File: Lab 9/t2.go
package main

import (
  "fmt"
  "os"
  "bufio"
  "strings"
)

func main() {
  fmt.Println("Please input a line:")
  line, e := bufio.NewReader(os.Stdin).ReadString('\n')
  if e != nil {
    fmt.Println("Read error:", e)
    return
  }

  fmt.Println("The first word is:", strings.Fields(line)[0])
}

File: Lab 9/t2.rs
fn main() {
  println!("Please input a line:");

  let mut buf = String::new();
  match std::io::stdin().read_line(&mut buf) {
    Ok(_) => {}
    Err(e) => {
      eprintln!("Read error: {}", e);
      std::process::exit(1);
    }
  }

  println!("The first word is: {}",
           buf.split_whitespace().next().unwrap_or(""));
}
